### PR TITLE
Use an enumeration for IdentifierType, rather than a CSV list

### PR DESCRIPTION
**Pull request by alexwlchan**
This makes it more consistent with some of our other models (LocationType, AccessStatus, etc.) and ensures a more efficient JSON serialisation, e.g.

```
{"id":"miro-image-number"}
```

rather than

```
{"id":"miro-image-number","label":"Miro image number","ontologyType":"IdentifierType"}
```

which is just generally efficient, and I'm keeping 🤞 might sort out the relation embedder by trimming the size of the Work JSON.

Closes https://github.com/wellcomecollection/platform/issues/5093

### CODE REVIEW

**Review by jamesgorrie — APPROVED** [review]
Superb! Something that's been on the pet peeve list for too long. Nice work.

File: common/internal_model/src/main/scala/weco/catalogue/internal_model/identifiers/IdentifierType.scala [hunk 1]
```diff
@@ -1,47 +1,106 @@
...
+  val values = findValues
+
+  implicit val locationTypeEncoder: Encoder[IdentifierType] =
+    Encoder.forProduct1("id")(_.id)
+
+  implicit val locationTypeDecoder: Decoder[IdentifierType] =
+    Decoder.forProduct1("id")(IdentifierType.withName)
+
+  def apply(id: String): IdentifierType =
+    IdentifierType.withName(id)
```
Comment (jamieparkinson) [inline common/internal_model/src/main/scala/weco/catalogue/internal_model/identifiers/IdentifierType.scala:25]: How is it working without `entryName` like the other types use? Does it secretly support an `id` field as well as `entryName`?(!) https://github.com/wellcomecollection/catalogue/blob/51ab8cf0288a9362ceec65a318cfe6766534142a/common/internal_model/src/main/scala/weco/catalogue/internal_model/work/Format.scala#L10

Comment (alexwlchan) [reply common/internal_model/src/main/scala/weco/catalogue/internal_model/identifiers/IdentifierType.scala:25]: The answer might be… it doesn't 🙈 

I wanted to get this up for review quickly, so I haven't run the tests yet

Comment (jamieparkinson) [reply common/internal_model/src/main/scala/weco/catalogue/internal_model/identifiers/IdentifierType.scala:25]: 😬 well, there's your solution if it doesn't

Comment (alexwlchan) [reply common/internal_model/src/main/scala/weco/catalogue/internal_model/identifiers/IdentifierType.scala:23]: Indeed, that seems to be important…

**Pull request merged**